float32 r_data
float32 r_time
float32 l_data
float32 l_time
